src/ProjectOrigin.WalletSystem.Server/Repositories/CertificateRepository.cs

```diff
@@ -13,34 +13,42 @@ public class CertificateRepository : ICertificateRepository
 {
     private IDbConnection _connection;
 
-    public CertificateRepository(IDbConnection connection)
-    {
-        this._connection = connection;
-    }
+    public CertificateRepository(IDbConnection connection) => this._connection = connection;
 
-    public Task InsertSlice(Slice newSlice)
+    public async Task InsertSlice(Slice newSlice)
     {
-        return _connection.ExecuteAsync(@"INSERT INTO Slices(Id, DepositEndpointId, DepositEndpointPosition, RegistryId, CertificateId, Quantity, RandomR, SliceState) VALUES (@id, @depositEndpointId, @depositEndpointPosition, @registryId, @certificateId, @quantity, @randomR, @sliceState)", new { newSlice.Id, newSlice.DepositEndpointId, newSlice.DepositEndpointPosition, newSlice.RegistryId, newSlice.CertificateId, newSlice.Quantity, newSlice.RandomR, newSlice.SliceState });
+        var registryId = await _connection.QuerySingleAsync<Guid>("SELECT Id FROM Registries WHERE Name = @registry", new { newSlice.Registry });
```

Comment:
I thought the goal was to drop the Registries table

Reply:
Well i concentrated in this PR simply to remove all references to it outside of the repository.
Then we can remove the table and denormalise all the data in another PR later, this already made quality of life a lot better.

Reply:
Cool